tests/dummy/app/templates/docs/glimmer-components.md

```diff
@@ -0,0 +1,109 @@
+
+A *very* simple Glimmer component which lets you change the count of a value might look like this:
+
+```htmlbars
+<button {{on "click" this.minus}}>&minus;</button>
+{{this.count}}
+<button {{on "click" this.plus}}>+</button>
+```
+
+```ts
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking'
+import { action } from '@ember/object';
+
+export default class Counter extends Component {
+  @tracked count = 0;
+  
+  @action
+  plus() {
+    this.count += 1;
+  }
+  
+  @action
+  minus() {
+    this.count -= 1;
+  }
+}
+```
+
+Notice that there are no type declarations here – but this *is* actually a well-typed component. The type of `count` is `number`, and if we accidentally wrote something like `this.count = null` the compiler would give us an error.
+
+## A component with arguments
+
+So far so good, but of course most components aren’t quite this simple! Instead, they’re invoked by other templates and they can invoke other components themselves in their own templates.
+
+Glimmer components can receive both *arguments* and *attributes* when they are invoked. When you are working with a component’s backing class, you have access to the arguments but *not* to the attributes. The arguments are passed to the constructor, and then available as `this.args` on the component instance afterward. Let’s imagine a component which just logs the names of its arguments when it is first constructed:
+
+```ts
+import Component from '@glimmer/component';
+
+export default class ArgsDisplay extends Component {
+  constructor(owner: unknown, args: {}) {
+    super(owner, args);
+
+    Object.keys(args).forEach(argName => {
+      console.log(argName);
+    });
+  }
+}
+```
+
+<aside>
+
+If you’re used to the classic Ember Object model, there are two important differences in the constructor itself:
+
+- we use `super` instead of `this._super`
+- we *must* call `super` before we do anything else
+
+</aside>
+
+Notice that we have to start by calling `super` with `owner` and `args`. This may be a bit different from what you’re used to in Ember or other frameworks, but is normal for sub-classes in TypeScript. If the compiler just accepted any `...arguments`, a lot of potentially *very* unsafe invocations would go through. So, instead of using `...arguments`, we explicitly pass the *specific* arguments and make sure their types match up with what the super-class expects.
+
+The types for `owner` here and `args` line up with what the `constructor` for Glimmer components expect. The `owner` is specified as `unknown` because this is a detail we explicitly *don’t* need to know about. The `args` are `object` because a Glimmer component *always* receives an object containing its arguments, even if the caller didn’t pass anything: then it would just be an empty object.
+
+The `args` are also available on `this`, just as you’d expect, so we could change our definition to return the names of the arguments from a getter:
```

Comment:
>  just as you’d expect

I'm not 100% sure that users will "just expect" this, given that if they've been working with ember components they have probably been using `this.foo` or the now deprecated `this.attrs.foo`. Do you think that readers will be better-served by treating `this.args` as a potentially new concept?

Reply:
I'm assuming here and throughout (per the `aside` at the top of the doc) that this is a Glimmer Components With TypeScript guide, *not* a Glimmer Components In General guide. I'd really prefer to avoid repeating the materials in the guides—but I think spots like this might be a good place to put in further links in the "Is this bit new to you? Check out *this* part of the Ember Guides" vein. I'll add something like that here and push it up.

Reply:
Sounds like you're on top of this. I also would like to see us avoid duplicate documentation. Maybe something like 
```suggestion
The `args` passed to a Glimmer Component [are available on `this`](https://github.com/glimmerjs/glimmer.js/blob/2f840309f013898289af605abffe7aee7acc6ed5/packages/%40glimmer/component/src/component.ts#L12), so we could change our definition to return the names of the arguments from a getter:
```

Reply:
I've committed that for now, though I may opt to revise to a link to the Octane guides/API docs rather than the implementation in a later pass!